### ISSUE TITLE
ActiveRecord: AUTO_INCREMENT on composite key can not dump 

### ISSUE THREAD

**Issue by dominion525** [rails/rails#30894]
I use  partition table on MySQL. Partition table  have composite key.
This column set `AUTO_INCREMENT`. 
but it not wrote `auto_increment: true` on db/`schema.rb`.

### Steps to reproduce

```
class CreateArticles < ActiveRecord::Migration[5.1]
  def up
    create_table :articles, id: false do |t|
      t.integer  :id,        null: false
      t.datetime :posted_at, null: false
      t.text     :content,   null: false
    end

    execute 'ALTER TABLE articles ADD PRIMARY KEY(id, posted_at);'
    execute 'ALTER TABLE articles MODIFY id BIGINT(20) NOT NULL AUTO_INCREMENT;'
  end
end
```

``` MySQL console
mysql> DESC articles;
+-----------+------------+------+-----+---------+----------------+
| Field     | Type       | Null | Key | Default | Extra          |
+-----------+------------+------+-----+---------+----------------+
| id        | bigint(20) | NO   | PRI | NULL    | auto_increment |
| posted_at | datetime   | NO   | PRI | NULL    |                |
| content   | text       | NO   |     | NULL    |                |
+-----------+------------+------+-----+---------+----------------+
```

### Expected behavior

``` db/schema.rb 
  create_table "articles", primary_key: ["id", "posted_at"], force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin" do |t|
    t.bigint "id", null: false, auto_increment: true
    t.datetime "posted_at", null: false
    t.text "content", null: false
  end
```

### Actual behavior

missing `auto_increment: true` ...

``` db/schema.rb 
  create_table "articles", primary_key: ["id", "posted_at"], force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin" do |t|
    t.bigint "id", null: false
    t.datetime "posted_at", null: false
    t.text "content", null: false
  end
```

### System configuration
**Rails version**:  5.1.4

**Ruby version**: 2.4.2

**MySQL version**: 2.4.2

**mysql2 gem version**: 0.4.9

**Comment by kamipo** [comment]
Thanks for reporting the issue. I've fixed it and backported in 8b6e694e5febdcb39f72d98ff9d5e7124cc0ca32.

**Comment by trungtq-0433** [comment]
@kamipo 
Excuse me, I got an error when I used command `bundle exec rake db:schema:load`
```
ActiveRecord::StatementInvalid: Mysql2::Error: Incorrect table definition; there can be only one auto column and it must be defined as a key: ALTER TABLE shopping_list_items DROP PRIMARY KEY
```
